src/server.c

```diff
@@ -4290,6 +4313,86 @@ const char *RESP3_TYPE_STR[] = {
     "null",
 };
 
+/* Must match replySchemaType */
+const char *REPLY_SCHEMA_TYPE_STR[] = {
+    "string",
+    "number",
+    "integer",
+    "object",
+    "array",
+    "boolean",
+    "null",
+    "unspecified", /* Not actually used... */
+};
+
+void addReplyCommandReplySchema(client *c, struct commandReplySchema *rs) {
+    long maplen = 0;
+    if (rs->type != SCHEMA_TYPE_UNSPECIFIED) maplen++;
+    if (rs->description) maplen++;
+    if (rs->notes) maplen++;
+    if (rs->uniqueItems) maplen++;
+    if (rs->additionalItems) maplen++;
+    if (rs->minItems) maplen++;
+    if (rs->maxItems) maplen++;
+    if (rs->items) maplen++;
+    if (rs->oneOf) maplen++;
+
+    addReplyMapLen(c, maplen);
+
+    if (rs->type != SCHEMA_TYPE_UNSPECIFIED) {
+        addReplyBulkCString(c, "type");
+        addReplyBulkCString(c, REPLY_SCHEMA_TYPE_STR[rs->type]);
+    }
+
+    if (rs->description) {
+        addReplyBulkCString(c, "description");
+        addReplyBulkCString(c, rs->description);
+    }
+
+    if (rs->notes) {
+        addReplyBulkCString(c, "notes");
+        addReplyBulkCString(c, rs->notes);
+    }
+
+    if (rs->uniqueItems) {
+        addReplyBulkCString(c, "uniqueItems");
+        addReplyBool(c, rs->uniqueItems);
+    }
+
+    if (rs->additionalItems) {
+        addReplyBulkCString(c, "additionalItems");
+        addReplyBool(c, rs->additionalItems);
+    }
+
+    if (rs->minItems) {
+        addReplyBulkCString(c, "minItems");
+        addReplyLongLong(c, rs->minItems);
+    }
+
+    if (rs->maxItems) {
+        addReplyBulkCString(c, "maxItems");
+        addReplyLongLong(c, rs->maxItems);
+    }
+
+    if (rs->items) {
+        addReplyBulkCString(c, "items");
+        if (rs->num_items == 1) {
+            addReplyCommandReplySchema(c, &rs->items[0]);
+        } else {
+            addReplyArrayLen(c, rs->num_items);
+            for (int i = 0; i < rs->num_items; i++)
+                addReplyCommandReplySchema(c, &rs->items[i]);
+        }
```

Comment:
here we assume that if there's only one item, it must be the schema to validate each item, and we do not support an array of items with size==1.

i.e. we can't tell the difference between
```
{
      "type": "array",
      "items": {
          "type":"string"
      }
 }
 ```
 
 and
 ```
 {
      "type": "array",
      "items": [
            {
                "type":"string"
            }
      ]
 }
 ```
 
 one might argue that the latter has no use in Redis (it means that the first item in the array is a string, and we don't care about the rest. if additionalItems=false is also provided, it means the array must have exactly one element, and it's a string. both cases seem to be not useful for Redis)